setup.py

```diff
@@ -1,39 +1,67 @@
-from setuptools import setup, find_packages
+"""setuptools config for nowplaypadgen"""
+
 import os
 import re
+import sys
+
+import nowplaypadgen
```

Comment:
Is this really safe? I guess this will fail, as soon as `nowplaypadgen/__init__.py` imports packages from `install_requires` dependencies (which aren't available yet). See also [Single-sourcing the package version](https://packaging.python.org/guides/single-sourcing-package-version/#single-sourcing-the-package-version) number 6. (set the value to `__version__` in `sample/__init__.py` and import `sample` in `setup.py`.)

Reply:
I think I prefer 4. in the linked article.

Reply:
Originally, I also favoured 4, but then decided to go with 1: Although, some regex parsing is necessary here, which definitely looks a bit scary and overcomplicated at first sight, the (external) parsing happens only once within `setup.py` and not during run time. Apart from that, `__version__` will be incorporated within the Python byte code (*.pyc), reading it will be much faster than parsing a `VERSION` file.

But anyway, my original comment wasn't about the preferred single-source method, but rather about the `import nowplaypadgen` statement, which could easily end up in a chicken-and-egg problem (due to not yet installed but imported dependencies, as mentioned within method 6.).

Reply:
Would that be a chicken-and-egg problem when installing an egg?

IMO you need to get the setup.py from somewhere and that somewhere bundles the nowplaying dir anyway. Importing it like this should work because `.` is in the default Python path.

Reply:
There seems to be a little misunderstanding here: My concerns (and those mentioned at number 6.) are not about the actual `nowplaypadgen` module itself, but rather about any (possible upcoming) module dependencies.

Imaging, that `setup.py` should install the module `foo` (`install_requires = ['foo']`), and `nowplaypadgen/__init__.py` loads `foo` (`import foo`). What will happen is that `setup.py` will fail as soon as it tries to import `nowplaypadgen` as the `foo` module is not available (installed) yet.

Reply:
I understand what you are getting at and ended up looking at how django [does](https://github.com/django/django/blob/9829b75d5bf468cb18e0fffdfc4a65dd7d8a6b2d/django/__init__.py) [this](https://github.com/django/django/blob/9829b75d5bf468cb18e0fffdfc4a65dd7d8a6b2d/django/utils/version.py#L5). Everything that has side effects depending on modules that are not available is kind of isolated by being wrapped in a setup function. 

While they don't use this pattern for they [description](https://github.com/django/django/blob/9829b75d5bf468cb18e0fffdfc4a65dd7d8a6b2d/setup.py#L41) they do use similar code to get [the version](https://github.com/django/django/blob/9829b75d5bf468cb18e0fffdfc4a65dd7d8a6b2d/setup.py#L32).

Anyway, this kind of depends on what exactly we'll be integrating this with so I'll move the string for now but would still prefer that we don't load any external modules in `__init__` without letting the user tell us so explicitly.

Reply:
Can you push this change please? I will merge the PR afterwards.